src/org/jitsi/impl/neomedia/transform/fec/FlexFec03Packet.java

```diff
@@ -0,0 +1,142 @@
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |k|                   Mask [15-45] (optional)                   |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |k|                                                             |
+ * +-+                   Mask [46-108] (optional)                  |
+ * |                                                               |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |                     ... next in SSRC_i ...                    |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ */
+
+public class FlexFec03Packet
+    extends RawPacket
+{
+    protected FlexFec03Header header;
+
+    public static FlexFec03Packet create(RawPacket p)
+    {
+        return create(p.getBuffer(), p.getOffset(), p.getLength());
+    }
+
+    public static FlexFec03Packet create(byte[] buffer, int offset, int length)
+    {
+        FlexFec03Packet flexFecPacket = new FlexFec03Packet(buffer, offset, length);
+        FlexFec03Header header = FlexFec03HeaderReader.readFlexFecHeader(
+            flexFecPacket.getBuffer(),
+            flexFecPacket.getFlexFecHeaderOffset(),
+            flexFecPacket.getLength() - flexFecPacket.getHeaderLength());
+        if (header == null)
+        {
+            return null;
+        }
+        flexFecPacket.header = header;
+        return flexFecPacket;
+    }
+
+    /**
+     * Ctor
+     * @param p a RawPacket representing a flex fec packet
+     */
+    private FlexFec03Packet(RawPacket p)
+    {
+        this(p.getBuffer(), p.getOffset(), p.getLength());
+    }
+
+    /**
+     * Ctor
+     * @param buffer rtp packet buffer
+     * @param offset offset at which the rtp packet starts in the given
+     * buffer
+     * @param length length of the packet
+     */
+    private FlexFec03Packet(byte[] buffer, int offset, int length)
+    {
+        super(buffer, offset, length);
+    }
+
+    /**
+     * Get the list of media packet sequence numbers protected by this
+     * FlexFec03Packet
+     * @return the list of media packet sequence numbers protected by this
+     * FlexFec03Packet
+     */
+    public List<Integer> getProtectedSequenceNumbers()
+    {
+        return this.header.protectedSeqNums;
+    }
+
+    /**
+     * Get the size of the flexfec header for this packet
+     * @return the size of the flexfec header for this packet
+     */
+    public int getFlexFecHeaderSize()
+    {
+        return this.header.size;
+    }
+
+    /**
+     * Get the media ssrc protected by this flexfec packet
+     * @return the media ssrc protected by this flexfec packet
+     */
+    public long getProtectedSsrc()
+    {
+        return this.header.protectedSsrc;
+    }
+
+    /**
+     * Returns the size of the FlexFEC payload, in bytes
+     * @return the size of the FlexFEC packet payload, in bytes
+     */
+    public int getPayloadLength()
```

Comment:
RawPacket already has a method with this signature. Can we add @Override, or rename it to getFlexFecHeaderLength() for example?

Reply:
will rename